File: src/main/java/com/msg/gauth/domain/email/presentation/EmailAuthController.java
package com.msg.gauth.domain.email.presentation;

import com.msg.gauth.domain.email.presentation.dto.request.EmailSendDto;
import com.msg.gauth.domain.email.services.MailSendService;
import com.msg.gauth.domain.email.services.MailVerificationService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/email")
@RequiredArgsConstructor
public class EmailAuthController {
    private final MailSendService mailSendService;
    private final MailVerificationService mailVerificationService;

    @PostMapping
    public ResponseEntity<Void> emailSend(@RequestBody EmailSendDto emailSendDto){
        mailSendService.execute(emailSendDto);
        return ResponseEntity.noContent().build();
    }

    @GetMapping("/authentication")
    public ResponseEntity<Void> emailVerification(@RequestParam String email, @RequestParam String uuid){
        mailVerificationService.execute(email, uuid);
        return ResponseEntity.noContent().build();
    }

}

File: src/main/kotlin/com/msg/gauth/domain/email/presentation/EmailAuthController.kt
package com.msg.gauth.domain.email.presentation

import com.msg.gauth.domain.email.presentation.dto.EmailSendDto
import com.msg.gauth.domain.email.services.MailSendService
import com.msg.gauth.domain.email.services.MailVerificationService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/email")
class EmailAuthController(
    private val mailSendService: MailSendService,
    private val mailVerificationService: MailVerificationService
) {
    @PostMapping
    fun emailSend(@RequestBody emailSendDto: EmailSendDto): ResponseEntity<Void> {
        mailSendService.execute(emailSendDto)
        return ResponseEntity.noContent().build()
    }

    @GetMapping("/authentication")
    fun emailVerification(@RequestParam email: String, @RequestParam uuid: String): ResponseEntity<Void> {
        mailVerificationService.execute(email, uuid)
        return ResponseEntity.noContent().build()
    }
}